registry/api/protocol/eureka/applications.go

```diff
@@ -14,7 +14,58 @@
 
 package eureka
 
+import (
+	"encoding/json"
+	"fmt"
+)
+
 // Applications is an array of application objects
 type Applications struct {
-	Application []*Application `json:"application,omitempty"`
+	VersionDelta int64
+	Hashcode     string
+	Application  []*Application
+}
+
+// This struct is used ONLY for the specific UnmarshalJSON
+type singleApplications struct {
+	VersionDelta int64        `json:"versions__delta,omitempty"`
+	Hashcode     string       `json:"apps__hashcode,omitempty"`
+	Application  *Application `json:"application,omitempty"`
+}
+
+// This struct is used ONLY for the specific UnmarshalJSON
+type multiApplications struct {
+	VersionDelta int64          `json:"versions__delta,omitempty"`
```

Comment:
Consider using an embedded struct to capture VersionDelta and Hashcode. See https://play.golang.org/p/A3Fv9WW9A5